numba_dpex/decorators.py

```diff
@@ -155,6 +155,7 @@ def dpjit(*args, **kws):
     kws.update({"nopython": True})
     kws.update({"parallel": True})
     kws.update({"pipeline_class": DpjitCompiler})
+    kws.update({"_target": "dpex"})
```

Comment:
I ran into a problem with this change previously (see the FIXME note few lines below). The `dpnp` overloads are right now added to the CPUTarget and changing dpjit to use `DpexTarget` makes numba not locate the dpnp overloads.
#1027 needs to be completed first before we can do the change.